contrib/win32/win32compat/ssh-agent/connection.c

```diff
@@ -136,26 +159,94 @@ get_con_client_type(struct agent_connection* con)
 	    GetTokenInformation(token, TokenUser, info, info_len, &info_len) == FALSE)
 		goto done;
 
-	sid_size = SECURITY_MAX_SID_SIZE;
-	if (CreateWellKnownSid(WinLocalSystemSid, NULL, system_sid, &sid_size) == FALSE)
-		goto done;
-	sid_size = SECURITY_MAX_SID_SIZE;
-	if (CreateWellKnownSid(WinNetworkServiceSid, NULL, ns_sid, &sid_size) == FALSE)
-		goto done;
-	sid_size = SECURITY_MAX_SID_SIZE;
-	if (CreateWellKnownSid(WinLocalServiceSid, NULL, ls_sid, &sid_size) == FALSE)
-		goto done;
+	/* check if its localsystem */
+	{
+		sid_size = SECURITY_MAX_SID_SIZE;
```

Comment:
you just need to call IsWellKnownSid(info->User.Sid, WinLocalSystemSid) to find out if it is localsystem. no need to create the sid and compare